tests/func/metrics/test_diff.py

```diff
@@ -153,3 +153,20 @@ def test_no_commits(tmp_dir):
     assert Git().no_commits
 
     assert Repo.init().metrics.diff() == {}
+
+
+def test_metrics_diff_dirty(tmp_dir, scm, dvc, run_copy_metrics):
```

Comment:
Need to add a "dirty" test for plots too. Did test it manually though. Will add one later in this PR.